libafl/src/fuzzer/mod.rs

```diff
@@ -612,14 +652,38 @@ where
         #[cfg(feature = "introspection")]
         state.introspection_monitor_mut().mark_manager_time();
 
+        let mut fuzzed_flag = false;
         {
             let mut testcase = state.testcase_mut(idx)?;
             let scheduled_count = testcase.scheduled_count();
-
+            if scheduled_count == 0 {
+                // Mark this testcase is fuzzed
+                fuzzed_flag = true;
+            }
             // increase scheduled count, this was fuzz_level in afl
             testcase.set_scheduled_count(scheduled_count + 1);
         }
 
+        if fuzzed_flag {
+            let pending_size = state.pending_mut();
+            if *pending_size > 0 {
+                *pending_size -= 1;
```

Comment:
Here it is

Reply:
ah ok.
but the problem is
1. it's better to put all these code to scheduler.
2. not every fuzzer wants this info. so you gotta let the user decide what scheduler they choose. then if they want this info, they choose the right scheduler

Reply:
or put it into the calibration stage.
it is where bitmap_size or exec_time, or etc.. are calculated.
it makes sense to do it for pending_favored, pending... too in calibration stage.